src/NServiceBus.Core/Transports/Msmq/ReceiveWithNoTransaction.cs

```diff
@@ -4,17 +4,21 @@ namespace NServiceBus
     using System.Collections.Generic;
     using System.Messaging;
     using System.Threading.Tasks;
-    using NServiceBus.Extensibility;
-    using NServiceBus.Logging;
-    using NServiceBus.Transports;
+    using Extensibility;
+    using Logging;
+    using Transports;
 
     class ReceiveWithNoTransaction : ReceiveStrategy
     {
         public override async Task ReceiveMessage(MessageQueue inputQueue, MessageQueue errorQueue, Func<PushContext, Task> onMessage)
         {
-
             var message = inputQueue.Receive(TimeSpan.FromMilliseconds(10), MessageQueueTransactionType.None);
 
+            if (message == null)
```

Comment:
I thought we discussed the null check is not necessary. At least you tried to convince me about that. If that is still true why was id added here?